src/MongoDB.Driver.Core/Core/Operations/AggregateToCollectionOperation.cs

```diff
@@ -288,5 +289,28 @@ private void EnsureIsOutputToCollectionPipeline()
                 throw new ArgumentException("The last stage of the pipeline for an AggregateOutputToCollectionOperation must have a $out or $merge operator.", "pipeline");
             }
         }
+
+        private IReadOnlyList<BsonDocument> SimplifyOutStageIfOutputDatabaseIsSameAsInputDatabase(IReadOnlyList<BsonDocument> pipeline)
+        {
+            var lastStage = _pipeline.Last();
```

Comment:
if I'm not mistaken there is no reason to use both a `pipeline` value as an method argument and `_pipeline` as a global field. If so, let's use only value from method argument

Reply:
This is a bug. Thanks for catching it.

Reply:
OK... so not actually a bug in the sense that it wasn't affecting the operation of the test, but it was incorrect and has been fixed.